pkg/web/transactions.go

```diff
@@ -592,6 +765,207 @@ func (s *Server) RejectTransaction(c *gin.Context) {
+		return
+	}
+
+	if ts := in.ParseTransferState(); ts != trisa.TransferStateUnspecified && ts != trisa.TransferReview {
+		c.JSON(http.StatusBadRequest, api.Error(fmt.Errorf("this endpoint does not accept the %q transfer state", ts.String())))
+		return
+	}
+
+	ctx := c.Request.Context()
+	log := logger.Tracing(ctx).With().Str("envelope_id", envelopeID.String()).Logger()
+
+	// Create a secure envelope with a Payload
+	if payload, err = in.Payload(); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.Error("could not create payload for transfer repair"))
+		return
+	}
+
+	// Send the payload with the accept transfer state
+	if packet, err = postman.Send(payload, envelopeID, trisa.TransferReview, log); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.Error("could not create outgoing packet for transfer repair"))
+		return
+	}
+
+	// Lookup the transaction from the database
+	if packet.Transaction, err = s.store.RetrieveTransaction(ctx, envelopeID); err != nil {
+		if errors.Is(err, dberr.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.Error("transaction not found"))
+			return
+		}
+
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error(err))
+		return
+	}
+
+	// Get the counterparty from the database
+	if packet.Counterparty, err = s.store.RetrieveCounterparty(ctx, packet.Transaction.CounterpartyID.ULID); err != nil {
+		if errors.Is(err, dberr.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.Error("counterparty not found: transaction needs to be updated"))
+			return
+		}
+
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error(err))
+		return
+	}
+
+	// Create a prepared transaction to update the transaction and secure envelopes
+	if packet.DB, err = s.store.PrepareTransaction(ctx, envelopeID); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error("unable to send transfer to remote counterparty"))
+		return
+	}
+	defer packet.DB.Rollback()
+
+	// Update the transaction based on the outgoing message from the API client
+	if err = packet.Out.UpdateTransaction(); err != nil {
+		c.Error(err)
+	}
+
+	// Send the secure envelope and get secure envelope response
+	// NOTE: SendEnvelope handles storing the incoming and outgoing envelopes in the database
+	if err = s.SendEnvelope(ctx, packet); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error("unable to send transfer to remote counterparty"))
+		return
+	}
+
+	// Update transaction state based on response from counterparty
+	if err = packet.In.UpdateTransaction(); err != nil {
+		c.Error(err)
+	}
+
+	if err = packet.DB.Commit(); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error("transfer sent but unable to store secure envelopes locally"))
+		return
+	}
+
+	// If the content requested is HTML (e.g. the web-front end), then redirect the user
+	// to the transaction detail page. and set a cookie to display a toast message
```

Comment:
If the user will be redirected, shouldn't there be an `htmx.Redirect` here? If so, we'll have to set a cookie like on line 413 in `SendEnvelopeForTransaction` instead of using `htmx.Trigger`. This is because the trigger event will not appear in the response once the user is redirected to the `detail` page.

Reply:
Ah, sorry I had just copied and pasted the comment and didn't read it through. I can certainly redirect if you prefer, I was just trying to do the same thing as `Accept` which uses the trigger; or I can just change the comment. What's your preference?